python/docs/code/python/index.rst

```diff
@@ -1,11 +1,1260 @@
+  0.334572E+5     0.334561E+5    0.337613E+5    0.329863E+5
+  0.3377042+5     0.330056E+5    0.319002E+5    0.323340E+5
+  0.340970E+5     0.3410972+5    0.342097E+5    0.338722E+5
+  ....
+
+Here we see that the name of keyword is 'PORV', and it consists of
+30000 elements of type REAL. At least all the file types summary
+(BASE.SMSPEC / BASE.UNSMRY / BASE.Snnnn), restart (BASE.UNRST /
+BASE.Xnnnn), grid (BASE.EGRID / BASE.GRID), rft (BASE.RFT) and init
+(BASE.INIT) consist of collections of such keywords. In the
+:code:`ert.ecl` code the classes :code:`EclKW` and :code:`Ecl3DKW` are
+designed to work with *one* such keyword.
+
+
+FortIO - work with binary Eclipse IO
+------------------------------------
+
+The binary files created by ECLIPSE are created according to a Fortran
+IO convention, which is a bit special, when writing e.g. a block of
+1000 bytes of data to disk the Fortran IO system will surround the
+block with a header and a tail denoting the size of the
+block. I.e. for the Fortran code:
+
+.. code:: bash
+
+   integer array(100)
+   write(unit) array
+
+What actually hits the disk looks like this:
+
+.. code:: bash
+
+   | 400 | array ...... | 400 |
+
+The header and tail is a 4 byte integer, which value is the number of
+bytes in the immediately following record. In addition the ECLIPSE
+files are ususally(?) written in the "the other" endianness. The
+:code:`FortIO` class should handle these matters transparently. For
+normal use of the library it should not be necessary to explicitly use
+the :code:`FortIO` class.
+
+The :code:`FortIO` class has quite good embedded documentation, and
+you are advised to use :code:`pydoc ert.ecl.FortIO` or browse the API
+documentation at :ref:`python_documentation` for further details.
+
+
+EclKW/Ecl3DKW - work with one keyword
+-------------------------------------
+
+The :code:`EclKW` class represents one keyword [1]_ from an Eclipse
+result file. The :code:`EclKW` class is essentially a vector of data,
+along with header with the size and type of data, and the name of the
+vector. Mostly you will get :code:`EclKW` instances by querying a
+:code:`EclFile` instance - but you can also instantiate
+:code:`EclKW` instances manually. The :code:`Ecl3DKW` class is for
+keywords which represent 3D properties like e.g. PRESSURE and PORO,
+this class requires a grid instance, and is documented with the
+:code:`EclGrid` documentation.
+
+
+Special methods - container
+...........................
+
+The :code:`EclKW` class implements the :code:`__getitem__()` and
+:code:`__setitem__()` methods which are used to implement access using
+the :code:`[ ]` notation, and the :code:`__len__()` method which gives
+the size of the :code:`EclKW` instance. In the example below we load
+an INIT file and extract the PERMX and PORO keywords, we then
+forcefully set permeability to zero for all elements where the
+porosity is below a limit:
+
+.. code:: python
+
+   from ert.ecl import EclFile,EclGrid
```

Comment:
And changing the `ert.ecl`, all of the imports in the examples must be changed as well
```
from ecl.file import EclFile
from ecl.grid import EclGrid
```